### PR TITLE
add persistence demonstration to SM example

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
doesn't change the flow, and doesn't implement actual persistence but shows how

### CODE REVIEW

File: BLE_SM/source/main.cpp [hunk 1]
```diff
@@ -396,21 +409,64 @@ class SMDeviceCentral : public SMDevice {
...
 {
     BLE& ble = BLE::Instance();
     events::EventQueue queue;
 
-    {
-        printf("\r\n PERIPHERAL \r\n\r\n");
-        SMDevicePeripheral peripheral(ble, queue, peer_address);
-        peripheral.run();
+    /* if filesystem creation fails or there is no filesystem the security manager
+     * will fallback to storing the security databse in memory */
```
Comment (pan-) [inline BLE_SM/source/main.cpp]: typo `databse` -> database

File: BLE_SM/source/main.cpp [hunk 2]
```diff
@@ -159,9 +162,19 @@ class SMDevice : private mbed::NonCopyable<SMDevice>,
             return;
         }
 
+        /* This path will be used to store bonding information but will fallback
+         * to storing in memory if file access fails (for example due to lack of a filesystem) */
+        const char* db_path = "/fs/bt_sec_db";
```
Comment (pan-) [inline BLE_SM/source/main.cpp]: May we use a configuration parameter ?

Comment (paul-szczepanek-arm) [reply BLE_SM/source/main.cpp]: you mean for the example or for the feature as a whole?

**Comment by paul-szczepanek-arm** [comment]
@pan- memory usage:
ram 512 bytes
flash 16 616 bytes

**Comment by MarceloSalazar** [comment]
Please note for OOB, we're working with mbed-os-5.9.0-oob branch.
If you think that branch should be updated to fix a critical issue, then retarget this PR. Thanks!

**Comment by AGlass0fMilk** [comment]
What is the status of this PR?

**Comment by donatieng** [comment]
Waiting for these to go in before merging: https://github.com/ARMmbed/mbed-os/pull/7681 ~https://github.com/ARMmbed/mbed-os/pull/7687~ https://github.com/ARMmbed/mbed-os/pull/7688

**Pull request merged**